### ISSUE TITLE
[Feature] combine --browser with channel in playwright config

### ISSUE THREAD

**Issue by bgotink** [microsoft/playwright#23840]
The `--browser` option can be used to override which browser(s) are used to run the tests, but this only works for non-Chromium browser types if the `channel` property is not set in the playwright config. Otherwise the tests fail with one of these two errors:

```
browserType.launch: Unsupported firefox channel "chrome"
browserType.launch: Unsupported webkit channel "chrome"
```

In another scenario where the `--browser` option doesn't work, an error is thrown by the CLI before the tests start executing:

```
Error: Cannot use --browser option when configuration file defines projects. Specify browserName in the projects instead.
    at new FullConfigInternal (/.../node_modules/@playwright/test/lib/common/config.js:56:63)
```

It would be an improvement if either the configured channel were ignored when the `--browser` option is used, or the `playwright test` command were to throw an error similar to the one above instead of trying to run the tests.

**Comment by dgozman** [comment]
@bgotink We recommend setting up projects in this case. The `--browser` flag is for getting started, but we expect most users to have a config with predefined projects. You can have a project for "chrome" channel, and another one for "chromium", [here is an example](https://playwright.dev/docs/test-projects#configure-projects-for-multiple-browsers).

When projects are set up, use `npx playwright test --project=chrome` instead of `npx playwright test --browser=chrome`.

Let me know whether this helps.

**Comment by bgotink** [comment]
@dgozman Yeah that would help, but that means developers have to first figure out why the tests fail to find this solution.

I've got a command that scaffolds a playwright config, similar to what `npm init playwright` does but specifically to add playwright to an existing angular application. The default playwright config it creates sets `channel: 'chrome'` as most people have chrome installed<sup>[[citation needed\]](https://en.wikipedia.org/wiki/Citation_needed)</sup> and by using the channel they can get started using `ng e2e` without having to learn about `playwright install` just yet.

If they then pass `ng e2e --browser chromium` everything works, but any of the other values causes problems. The fact that these problems are shown as failing tests rather than an error about an invalid option being passed, is confusing.

**Comment by dgozman** [comment]
> `ng e2e --browser chromium`

If you are wrapping Playwright, I'd recommend making your own command-line arguments and translating them into `--project` argument to Playwright. This would be easier for everyone, because you won't depend on Playwright's decisions about the command line.

Alternatively, you can not use `channel` and make sure your scaffolding downloads Playwright's browsers. Or just teach your users to use proper Playwright. Overall, this does not seem like a feature we would really want in Playwright itself, and in such cases wrapping is usually the better way.

> The fact that these problems are shown as failing tests rather than an error about an invalid option being passed, is confusing.

I totally agree with this one! Perhaps we can do something here, we'll take a look.

**Comment by pavelfeldman** [comment]
#### Why was this issue closed?

Thank you for your involvement. This issue was closed due to limited engagement (upvotes/activity), lack of recent activity, and insufficient actionability. To maintain a manageable database, we prioritize issues based on these factors.

If you disagree with this closure, please open a new issue and reference this one. More support or clarity on its necessity may prompt a review. Your understanding and cooperation are appreciated.
